batch_job/src/batch_task.c

```diff
@@ -4,9 +4,23 @@ This software is distributed under the GNU General Public License.
 See the file COPYING for details.
 */
 
+#include <assert.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <string.h>
+#include <errno.h>
+
 #include "batch_task.h"
 #include "stringtools.h"
 #include "xxmalloc.h"
+#include "path.h"
+#include "debug.h"
+
+struct batch_task_wrapper {
+	struct list *pre;
+	struct list *post;
+	struct list *cmd;
```

Comment:
Should a batch_task_wrapper contain a batch_task_wrapper to allow for nesting? Also how do we want to handle wrapping? We can talk more in depth later, but my initial thinking on this code is that we have functionality like a linked list for wrappers.